services/venice-server/src/main/java/com/linkedin/venice/stats/ServerConnectionStats.java

```diff
@@ -0,0 +1,43 @@
+package com.linkedin.venice.stats;
+
+import io.tehuti.metrics.MetricsRepository;
+import io.tehuti.metrics.Sensor;
+import io.tehuti.metrics.stats.Max;
+
+
+public class ServerConnectionStats extends AbstractVeniceStats {
+  public static final String ROUTER_CONNECTION_COUNT_GAUGE = "router_connection_count_gauge";
+  public static final String CLIENT_CONNECTION_COUNT_GAUGE = "client_connection_count_gauge";
+
+  private final Sensor routerConnectionCountSensor;
+  private final Sensor clientConnectionCountSensor;
+
+  private int routerConnectionCount;
+  private int clientConnectionCount;
+
+  public ServerConnectionStats(MetricsRepository metricsRepository, String name) {
+    super(metricsRepository, name);
+    routerConnectionCountSensor = registerSensorIfAbsent(ROUTER_CONNECTION_COUNT_GAUGE, new Gauge(), new Max());
```

Comment:
Also I wonder whether the default constructor for `Gauge` works at all.
Normally, we will offer a `Measurable` callback when constructing a `Gauge` metric and for this use case, it will be:
```
new Gauge(() -> routerConnectionCount)
```

Also `Max` is a window-based stat, and if there is no update of the sensor in the past 1 min, the `max` value will be `0`, which will be confusing.

If we would like to use `Max` to capture the conn storm issue, we can potentially use `OccurrenceRate` stat to achieve the same goal.

Reply:
Oh, maybe that's why the gauge is returning 0 in the integration test. Initially I thought it's due to connections closing already. Let me fix this.

Reply:
Update: the gauge is indeed working and it's returning 0 in the integration test because the channel is actually unregistered. Verified this by commenting out the decrement methods.